res/wrap/wrap.jl

```diff
@@ -1,511 +1,219 @@
-    added = setdiff(keys(new_defs), keys(existing_defs))
-    if !isempty(added)
-        open(existing_output_file, "a") do io
-            for fn in added
-                pos = new_defs[fn]
-                text = new_output_text[pos.offset+1:pos.offset+pos.span]
-                println(io)
-                println(io, text)
-                @warn "Adding definition of $fn, please review!"
+            # insert `@checked` before each function with a `ccall` returning a checked type`
+            rettyp = call_expr.args[2]
+            checked_types = if haskey(options, "api")
+                get(options["api"], "checked_rettypes", Dict())
+            else
+                String[]
+            end
+            if rettyp isa Symbol && String(rettyp) in checked_types
+                node.exprs[1] = Expr(:macrocall, Symbol("@checked"), nothing, expr)
             end
         end
     end
-
-    # apply the common file
-    if name == "custatevec"
-        cp(new_common_file, joinpath(dirname(dirname(@__DIR__)) , "lib", "CUSTATEVEC", "src", basename(new_common_file)); force=true)
-    elseif name == "cutensornet"
-        cp(new_common_file, joinpath(dirname(dirname(@__DIR__)) , "lib", "CUTENSORNET", "src", basename(new_common_file)); force=true)
-    else
-        cp(new_common_file, joinpath(dirname(dirname(@__DIR__)) , "lib", name, basename(new_common_file)); force=true)
-    end
-
-    return
 end
 
-function main(; name="all")
+function main(name="all")
     cuda = joinpath(CUDA_full_jll.artifact_dir, "cuda", "include")
     cupti = joinpath(CUDA_full_jll.artifact_dir, "cuda", "extras", "CUPTI", "include")
     cudnn = joinpath(CUDNN_jll.artifact_dir, "include")
     cutensor = joinpath(CUTENSOR_jll.artifact_dir, "include")
     cuquantum = joinpath(cuQuantum_jll.artifact_dir, "include")
+    opengl = joinpath(Libglvnd_jll.artifact_dir, "include")
 
-    (name == "all" || name == "cudadrv") && process("cudadrv", "$cuda/cuda.h","$cuda/cudaGL.h","$cuda/cudaProfiler.h";
-            include_dirs=[cuda], libname="cuda")
+    if name == "all" || name == "cudadrv"
+        wrap("cuda", ["$cuda/cuda.h","$cuda/cudaGL.h","$cuda/cudaProfiler.h"];
+            include_dirs=[cuda, opengl])
+    end
 
-    (name == "all" || name == "nvtx") && process("nvtx", "$cuda/nvtx3/nvToolsExt.h", "$cuda/nvtx3/nvToolsExtCuda.h";
+    if name == "all" || name == "nvtx"
+        wrap("nvtx", ["$cuda/nvtx3/nvToolsExt.h", "$cuda/nvtx3/nvToolsExtCuda.h"];
             include_dirs=[cuda])
+    end
 
-    (name == "all" || name == "nvml") && process("nvml", "$cuda/nvml.h"; include_dirs=[cuda])
+    if name == "all" || name == "nvml"
+        wrap("nvml", ["$cuda/nvml.h"]; include_dirs=[cuda])
+    end
 
-    (name == "all" || name == "cupti") && process("cupti", "$cupti/cupti.h", "$cupti/cupti_profiler_target.h";
+    if name == "all" || name == "cupti"
+        # NOTE: libclang (the C API) doesn't support/expose the __packed__/aligned attributes,
```

Comment:
The latest generator supports packed/aligned structs/unions by overloading field accessing methods.

Reply:
Great!